lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
-
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+
+    if @front == @back
+      raise Error, "Queue is full!"
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
+
+    if @store[@front].nil?
+      @front = (@front + 1) % @store.length
+    end
```

Comment:
You're doing this part here because of how you set front and back to 0 and 1 when the queue is empty.  You can fix it by adjusting the two ifs into an if and an elsif.